simply/scenario.py

```diff
@@ -171,16 +250,19 @@ def load(dirpath, data_format):
 
 def create_random(num_nodes, num_actors, weight_factor):
     pn = power_network.create_random(num_nodes)
-    actors = [actor.create_random("H" + str(i)) for i in range(num_actors)]
-
     # Add actor nodes at random position (leaf node) in the network
     # One network node can contain several actors (using random.choices method)
-    map_actors = pn.add_actors_random(actors)
+
     # Update shortest paths and the grid fee matrix
     pn.update_shortest_paths()
     pn.generate_grid_fee_matrix(weight_factor)
-
-    return Scenario(pn, actors, map_actors)
+    mm_buy_prices = np.random.random(100)
+    scenario = Scenario(pn, [], None, buy_prices=mm_buy_prices)
+    environment = scenario.environment
```

Comment:
if environment has to be handed to actors, it makes sense to make those inputs (actors, actor map) optional

idea: define a function add_actors(actors, actor_map) in Scenario class, where the actors get the environment and are also added to the environment.

Reply:
I think we can leave out actors from init all the way, since actors need a reference to the scenario.environment